stcgal/frontend.py

```diff
@@ -103,45 +114,57 @@ def load_file_auto(self, fileobj):
     def program_mcu(self):
         """Execute the standard programming flow."""
 
-        code_size = self.protocol.model.code
-        ee_size = self.protocol.model.eeprom
+        if self.opts.option: self.emit_options(self.opts.option)
+
+        if self.protocol.split_code and self.protocol.model.iap:
+            code_size = self.protocol.split_code
+            ee_size = self.protocol.split_eeprom
+        else:
+            code_size = self.protocol.model.code
+            ee_size = self.protocol.model.eeprom
 
         print("Loading flash: ", end="")
         sys.stdout.flush()
         bindata = self.load_file_auto(self.opts.code_image)
+        
+        if self.protocol.model.mcs251 and self.hexFileType != 32:
```

Comment:
Maybe I'm misunderstanding that, but isn't this too strict? If I understand right, because MCS-251 supports more than 64 KiB flash size, you may need a 32 bit length capable HEX file if all of the flash is used. But why does it need to be enforced always?

Furthermore, as far as I can tell `linearBaseAddress` is not actually used in the protocol code. Is something missing here?

Reply:
I don't remember the exact details, but the hex file format for an MCS-251 differs from that of an MCS-51. If you want to flash an MCS-251, not providing an adequate file is an error, it's not just the file format, the content is also different (MCS-251 has more instructions and its memory models are different).

Removed linearBaseAddress from frontend.py.